include/net/sntp_sock.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_SNTP_SOCK_H_
+#define ZEPHYR_INCLUDE_NET_SNTP_SOCK_H_
+
+#include <net/socket.h>
+
+struct sntp_ctx;
+
+/**
+ * @typedef sntp_resp_cb_t
+ * @brief SNTP response callback.
+ *
+ * @details The callback is called after a sntp response is received
+ *
+ * @param ctx Address of sntp context.
+ * @param status Error code of sntp response.
+ * @param epoch_time Seconds since 1 January 1970.
+ * @param user_data The user data given in sntp_request().
+ */
+
+typedef void (*sntp_resp_cb_t)(struct sntp_ctx *ctx,
+			       int status,
+			       u64_t epoch_time,
+			       void *user_data);
+
+/** SNTP context */
+struct sntp_ctx {
+	struct {
+		struct pollfd fds[1];
+		int nfds;
+		int fd;
+	} sock;
+
+	/** Timestamp when the request is departed the client for the server.
+	 *  This is used to check if the originated timestamp in the server
+	 *  reply matches the one in client request.
+	 */
+	u32_t		    expected_orig_ts;
+
+	/** SNTP response callback */
+	sntp_resp_cb_t      cb;
+
+	/** The user data of SNTP response callback */
+	void                *user_data;
+
+	/** Is this context setup or not */
+	bool                is_init;
```

Comment:
I don't think you need that, if it's not initialized fd will be invalid and sock operation should fail.